src/services/publisher_service/mod.rs

```diff
@@ -43,10 +45,18 @@ pub async fn start(
     metrics: Option<Metrics>,
     analytics: NotifyAnalytics,
 ) -> Result<(), sqlx::Error> {
-    let mut pg_notify_listener = PgListener::connect_with(&postgres).await?;
-    pg_notify_listener
-        .listen("notification_for_delivery")
-        .await?;
+    // Spawning a new task to observe messages queue stats by polling and export them to metrics
+    if let Some(metrics) = metrics.clone() {
+        tokio::spawn({
+            let postgres = postgres.clone();
+            async move {
+                loop {
+                    helpers::update_metrics_on_queue_stats(&metrics, &postgres).await;
+                    tokio::time::sleep(QUEUE_STATS_POLLING_INTERVAL).await;
```

Comment:
non-blocking: Might be nicer to use an [interval](https://docs.rs/tokio/latest/tokio/time/fn.interval.html) here instead of sleeping

Reply:
Don't mind changing this. Done.